ui/src/chat/ChatScroller/ChatScroller.tsx

```diff
@@ -184,6 +187,19 @@ export default function ChatScroller({
         data: keys,
       });
     }
+
+    // Sometimes the virtuoso component doesn't scroll to the bottom when
+    // switching chats. Diff remains zero when it shouldn't.
+    // This is a hack to force it to scroll to the bottom.
+
+    if (indexData.firstItemIndex === FIRST_INDEX && diff === 0) {
+      // We need to wait to make sure the virtuoso component has been updated.
+      setTimeout(() => {
+        virtuoso?.current?.scrollToIndex({
+          index: indexData.data.length - 1,
+        });
+      }, 50);
+    }
```

Comment:
did moving this into the `oldWhom !== whom` statement not work? I guess the index data wouldn't be accurate...

Reply:
Moving what into the oldWhom !== whom statement? Like moving this inside that if statement:

```
    const diff = mess.size - indexData.data.length;
    if (diff !== 0) {
      setIndexData({
        firstItemIndex: indexData.firstItemIndex - diff,
        data: keys,
      });
    }
```

No, doesn't work.